sdv/single_table/base.py

```diff
@@ -107,6 +112,16 @@ def __init__(self, metadata, enforce_min_max_values=True, enforce_rounding=True,
         self._fitted_sdv_version = None
         self._fitted_sdv_enterprise_version = None
         self._synthesizer_id = generate_synthesizer_id(self)
+        if self.table_name is None:
```

Comment:
My only fear is if somehow there's another way the table name ends up none. As long as that's impossible, this seems good

Reply:
Well we need to make sure that multi table synthesizers flag this correctly. As you can see on the changes done to `HMASynthesizer`, we should do the same on `HSASynthesizer` and others.

Reply:
After discussing adding a `flag` that is not in the `__init__` is to over complicate the current creation of the synthesizers and we are rolling back to the previous solution with the `disable_single_table_logging`.